examples/quantization_aware_training/torch/resnet18/main.py

```diff
@@ -0,0 +1,328 @@
+    )
+
+    return train_loader, val_loader, calibration_dataset
+
+
+def prepare_tiny_imagenet_200(dataset_dir: Path):
+    # Format validation set the same way as train set is formatted.
+    val_data_dir = dataset_dir / "val"
+    val_images_dir = val_data_dir / "images"
+    if not val_images_dir.exists():
+        return
+
+    val_annotations_file = val_data_dir / "val_annotations.txt"
+    with open(val_annotations_file, "r") as f:
+        val_annotation_data = map(lambda line: line.split("\t")[:2], f.readlines())
+    for image_filename, image_label in val_annotation_data:
+        from_image_filepath = val_images_dir / image_filename
+        to_image_dir = val_data_dir / image_label
+        if not to_image_dir.exists():
+            to_image_dir.mkdir()
+        to_image_filepath = to_image_dir / image_filename
+        from_image_filepath.rename(to_image_filepath)
+    val_annotations_file.unlink()
+    val_images_dir.rmdir()
+
+
+def run_benchmark(model_path: str, shape: List[int]) -> float:
+    command = f"benchmark_app -m {model_path} -d CPU -api async -t 15"
+    command += f' -shape "[{",".join(str(x) for x in shape)}]"'
+    cmd_output = subprocess.check_output(command, shell=True)  # nosec
+    match = re.search(r"Throughput\: (.+?) FPS", str(cmd_output))
+    return float(match.group(1))
+
+
+def get_model_size(ir_path: str, m_type: str = "Mb") -> float:
+    xml_size = os.path.getsize(ir_path)
+    bin_size = os.path.getsize(os.path.splitext(ir_path)[0] + ".bin")
+    for t in ["bytes", "Kb", "Mb"]:
+        if m_type == t:
+            break
+        xml_size /= 1024
+        bin_size /= 1024
+    model_size = xml_size + bin_size
+    return model_size
+
+
+def main():
+    torch.manual_seed(0)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    print(f"Using {device} device")
+
+    ###############################################################################
+    # Step 1: Prepare model and dataset
+    print(os.linesep + "[Step 1] Prepare model and dataset")
+    model = get_resnet18_model(device)
+
+    # Define loss function (criterion) and optimizer.
+    criterion = nn.CrossEntropyLoss().to(device)
+
+    model, acc1_fp32 = load_checkpoint(model)
+
+    train_loader, val_loader, calibration_dataset = create_data_loaders()
+
+    def transform_fn(data_item):
+        return data_item[0].to(device)
+
+    quantization_dataset = nncf.Dataset(calibration_dataset, transform_fn)
+
+    ###############################################################################
+    # Step 2: Quantize model
+    print(os.linesep + "[Step 2] Quantize model")
+
+    quantized_model = nncf.quantize(model, quantization_dataset)
+    acc1_int8_init = validate(val_loader, quantized_model, device)
+
+    print(f"Accuracy@1 of initialized INT8 model: {acc1_int8_init:.3f}")
+
+    compression_lr = 1e-5
+    optimizer = torch.optim.Adam(quantized_model.parameters(), lr=compression_lr)
```

Comment:
Move it to the fine tune quantized model step.

Reply:
Done